test/e2e/delete_test.go

```diff
@@ -126,20 +129,18 @@ func testWindowsNodeDeletion(t *testing.T) {
 			break
 		}
 	}
-
-	require.NotNil(t, windowsMachineSetWithLabel, "could not find MachineSet with Windows label")
-
-	// Scale the Windows MachineSet to 0
-	expectedNodeCount := int32(0)
-	windowsMachineSetWithLabel.Spec.Replicas = &expectedNodeCount
-	_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(),
-		windowsMachineSetWithLabel, meta.UpdateOptions{})
-	require.NoError(t, err, "error updating Windows MachineSet")
-
-	// we are waiting 10 minutes for all windows machines to get deleted.
-	err = testCtx.waitForWindowsNodes(expectedNodeCount, true, false, false)
-	require.NoError(t, err, "Windows node deletion failed")
-
+	// skip the scale down step if there is no MachineSet with Windows label
+	if windowsMachineSetWithLabel != nil {
+		// Scale the Windows MachineSet to 0
```

Comment:
This comment was more meaningful where we used to set `expectedNodeCount := int32(0)` next line. It would be good if you can move it back here.